BXMSDOC-5711: Release notes docs for DM and PAM 7.8 (#79)
kaldesai/kie-docs: doc-content/enterprise-only/release-notes/rn-known-issues-ref.adoc

Workaround: Update zip file by adding `dashbuilder-runtime.war.dodeploy` file as the copy of the folder `dashbuilder-runtime.war`. Alternatively, the distribution must be a war file and you can directly copy to the deployments folder.

.When you export the dashbuilder related data, gradual export displays internal data sources [https://issues.redhat.com/browse/RHPAM-3021[RHPAM-3021]]

Issue: When you export the dashbuilder related data, gradual export displays internal data sources for user selection. In fact, it should display the public data sets only.

Workaround: None.

endif::[]

.In Guided decision table, the verifier feature does not work properly for columns with the `Date` data type [https://issues.redhat.com/browse/RHDM-1399[RHDM-1399]]

Issue: In Guided decision table, the verifier feature throws an error when you open a table that contains a specific combination of template keys in the actions column. These template keys are associated with the `Date` data type.

Workaround: None.

> .You can not select the cell of the guided decision table [https://issues.redhat.com/browse/RHDM-1398[RHDM-1398]]

Reviewer: ```suggestion
.You cannot select the cell of a guided decision table [https://issues.redhat.com/browse/RHDM-1398[RHDM-1398]]
```

Issue: You can not select the data cell of the guided decision table with the help of a mouse.

Steps to reproduce:

* In Business Central, go to *Menu → Design → Projects* and click *Mortgages*.
* Open the `Pricing loans` guided decision table.
* Click the data cell of the guided decision table.

Expected result: You can select the data cell of the guided decision table with the help of a mouse.

> Actual result: You can not select the data cell of the guided decision table with the help of a mouse.

Reviewer: ```suggestion
Actual result: You cannot select the data cell of a guided decision table with the help of a mouse.
```

Workaround: You can select the header cell using mouse and then navigate into the data cells with help of keyboard arrow keys.

> .You can not use undo and redo keyboard shortcut keys in the grid editor [https://issues.redhat.com/browse/RHDM-1397[RHDM-1397]]

Reviewer: ```suggestion
.You cannot use undo and redo keyboard shortcut keys in the grid editor [https://issues.redhat.com/browse/RHDM-1397[RHDM-1397]]
```

> Issue: You can not perform undo (Ctrl+Z) or redo (Ctrl+Shift+Z) operation in the grid editor with the help of keyboard shortcut keys.

Author: Thank you so much for introducing me to this (kbd:[Ctrl+Z]) feature. I was not aware of it.

Steps to reproduce:

* Create a new DMN diagram.
* Add two decision nodes.
* Press Ctrl+Z.
+
One decision node is removed.
* Navigate to the grid editor of the remaining node.
* Add some context entries.
* Press Ctrl+Z.
+
The same context is present.

Expected result: You can use undo and redo keyboard shortcut keys in the grid editor

Actual result: You can not use undo and redo keyboard shortcut keys in the grid editor

Workaround: The *Undo* button from toolbar works correctly.

> .You can not convert the rule names from guided decision table into spreadsheet decision table [https://issues.redhat.com/browse/RHDM-1396[RHDM-1396]]

Reviewer: ```suggestion
.You cannot convert the rule names from guided decision table into a spreadsheet decision table [https://issues.redhat.com/browse/RHDM-1396[RHDM-1396]]
```

> Issue: You can not convert the rule names from guided decision table into spreadsheet decision table (XLS).

Reviewer: ```suggestion
Issue: You cannot convert the rule names from a guided decision table into spreadsheet decision table (XLS).
```

Steps to reproduce:

* In Business Central, go to *Menu → Design → Projects* and click *Mortgages*.
* Open the `Pricing loans` guided decision table.
* In *Columns* tab, display the rule name column.
* In *Model* tab, set some custom rule names.
* Save the guided decision table.
* Convert the guided decision table into `XLS` file via toolbar button.
* Open the generated `XLS` file, click *Source* tab, and verify the rule names.

Expected result: Custom rule names are correctly converted from guided decision table into spreadsheet decision table (XLS).

Actual result: Custom rule names are not correctly converted from guided decision table into spreadsheet decision table (XLS).

Workaround: None.

ifdef::DM[]

.Custom export functionality is not supported in the {PRODUCT} [https://issues.redhat.com/browse/RHDM-1402[RHDM-1402]]

> Issue: Custom export functionality requires at least one page to export the dashbuilder related data. In {PRODUCT}, you can not create any custom pages.

Reviewer: ```suggestion
Issue: Custom export functionality requires at least one page to export the dashbuilder related data. In {PRODUCT}, you cannot create any custom pages.
```

Workaround: None.

endif::[]

ifdef::PAM[]
== Process Designer
.When you create a text annotation from main palette, you receive a system error message [https://issues.redhat.com/browse/RHPAM-3052[RHPAM-3052]]

Issue: When you try to create a text annotation from main palette, you receive a system error message

Steps to reproduce:

* Create a process.
* Drag and drop a text annotation from main palette without expanding the palette.

Expected result: Text annotation is created without any system error.

Actual result: Text annotation is not created.

Workaround: Create a text annotation from expanded palette only.

> .When you morph to an event gateway, you can not delete the connected nodes and save the process [https://issues.redhat.com/browse/RHPAM-3036[RHPAM-3036]]

Reviewer: ```suggestion
.When you morph to an event gateway, you cannot delete the connected nodes and save the process [https://issues.redhat.com/browse/RHPAM-3036[RHPAM-3036]]
```

Issue: When you morph to an event gateway, you can not delete the connected nodes and save the process. Also, it is not possible to delete the sequence flows. You receive an error message.

Steps to reproduce:

* Create a process with a parallel gateway.
* Append any node expect Timer and Text Annotation.
* Morph that gateway to event gateway.
* Try to delete any connected nodes, sequence flow or the gateway itself.
* Save the process.

Expected result: You can delete the connected nodes, sequence flow. You can save the process. Also, you receive a warning message in case the process is not executable.

> Actual result: You can not delete the connected nodes, sequence flow. Also, you can not save the process.

Reviewer: ```suggestion
Actual result: You can not delete the connected nodes, sequence flow, and save the process.
```

Workaround: Change a gateway type back to its previous gateway type using morphing toolbox menu. For example, try to change the event gateway back to parallel gateway.
endif::[]

== Installer

.The {PRODUCT} installer contains references to {JWS} 5.2 [https://issues.redhat.com/browse/RHPAM-3077[RHPAM-3077]]

Issue: When you run the {PRODUCT} installer, the installation path step references to {JWS} 5.2 instead of {JWS} {JWS_VERSION}.

Workaround: None.

== DMN Designer

.When you include a PMML 4.4 model into a DMN model and then save or validate the model, DMN throws an error [https://issues.redhat.com/browse/RHDM-1400[RHDM-1400]]

> Issue: You can not include a PMML 4.4 model into a DMN model, DMN model throws an error.

Reviewer: ```suggestion
Issue: You cannot include a PMML 4.4 model into a DMN model, the DMN model throws an error.
```

Steps to reproduce:

* Create a new empty project in {CENTRAL}.
* Create an empty DMN model.
* Import one PMML asset with the 4.4 version into the project.
* Open the DMN model.
* Select the *Included Models* tab.
* Click the *Include Model* and select the PMML 4.4 model.
* Click *Include*.

> 

Reviewer: I'd suggest using a numbered list instead of bullets since these are sequential steps.

Expected result: You can include a PMML 4.4 model into a DMN model.

> Actual result: You can not include a PMML 4.4 model into a DMN model.

Reviewer: ```suggestion
Actual result: You cannot include a PMML 4.4 model into a DMN model.
```

Workaround: None

.When you try to invoke a Business Knowledge Model (BKM) function from included DMN model, the validation fails [https://issues.redhat.com/browse/RHDM-1395[RHDM-1395]]

> Issue: You can not call the BKM function imported from another DMN model, which is aliased with the name containing `.` characters. If you try to invoke that function, the validation fails.

Reviewer: ```suggestion
Issue: You can not call the BKM function imported from another DMN model, which is aliased with the name containing `.` characters. If you try to invoke BKM function, the validation fails.
```

Steps to reproduce:

* Create an empty `Salary.dmn` DMN model.
* Create an another empty `WorkingHours.dmn` DMN model.
* Import `WorkingHours.dmn` DMN model into `Salary.dmn` DMN model.
* Set an alias for `WorkingHours.dmn` model by including `.` character. For example, `working.hours.model`.

> * Validate the model. You receive an error message.

Reviewer: I'd suggest using a numbered list instead of bullets since these are sequential steps.

> * Validate the model. You receive an error message.

Reviewer: _You receive an error message_ should not be the part of the step. 
```suggestion
. Validate the model. 

You receive an error message.
```

Workaround: None

== {PLANNER}

.In Constraint Streams API, the performance of incremental score calculations is degraded using `groupBy()` method [https://issues.redhat.com/browse/RHDM-1385[RHDM-1385]]

> Issue: When you use the `groupBy()` method in Constraint Streams API, the the performance of incremental score calculations is degraded on high level.

Reviewer: ```suggestion
Issue: When you use the `groupBy()` method in Constraint Streams API, the performance of incremental score calculations is degraded on a high level.
```

Workaround: You can use the DRL score calculations.

== {OPENSHIFT}

ifdef::PAM[]

> .You can not create custom database image for PostgreSQL using PostgreSQL enterprise instance [https://issues.redhat.com/browse/RHPAM-3005[RHPAM-3005]]

Reviewer: ```suggestion
.You cannot create custom database image for PostgreSQL using PostgreSQL enterprise instance [https://issues.redhat.com/browse/RHPAM-3005[RHPAM-3005]]
```

Issue: When you create a custom database for PostgreSQL and push the image using PostgreSQL enterprise instance, {KIE_SERVER} fails to start.

Steps to reproduce:

> * Create a custom database and make the image build for PostgreSQL.

Reviewer: I'd suggest using a numbered list instead of bullets since these are sequential steps.

+
----
make build postgresql
----

* Push the custom database image using PostgreSQL enterprise instance.
+
----
db.primary_label=postgresplus116
db.port=5432
server_label_primary=postgresplus116
hibernate.dialect=org.hibernate.dialect.PostgresPlusDialect
hibernate.connection.url=jdbc\:edb\://edb-as11.mwqe.upshift.rdu2.redhat.com\:5432/dballo06
hibernate.connection.schema=public
datasource.class.xa=com.edb.xa.PGXADataSource
----

* Deploy the {KIE_SERVER} using S2I.

Expected result: You can create custom database image for PostgreSQL using PostgreSQL enterprise instance.

> Actual result: You can not create custom database image for PostgreSQL using PostgreSQL enterprise instance.

Reviewer: ```suggestion
Actual result: You cannot create a custom database image for PostgreSQL using PostgreSQL enterprise instance.
```

Workaround: None.

endif::[]

> .The {CENTRAL} pod fails to start when the `known_host` file is missing from the githook secret [https://issues.redhat.com/browse/RHPAM-3056[RHPAM-3056]]

Reviewer: ```suggestion
. {CENTRAL} pod fails to start when `known_host` file is missing from the Git hook secret [https://issues.redhat.com/browse/RHPAM-3056[RHPAM-3056]]
```

> Issue: The {CENTRAL} pod fails to start and remains stuck at *Container Creating* status if the `known_host` file is missing from the githook secret.

Reviewer: ```suggestion
Issue: The {CENTRAL} pod fails to start and remains stuck at *Container Creating* status if the `known_host` file is missing from the Git hook secret.
```

Steps to reproduce:

> * Create the following post-commit script.

Reviewer: I'd suggest using a numbered list instead of bullets since these are sequential steps.

+
----
cat post-commit
#!/usr/bin/sh
echo "running post-commit git hook"
git clone git@github.com:Sgitario/kjar-examples.git
----

* Upload this post-commit file into the Openshift.
+
----
oc create configmap githook-post-commit --from-file=post-commit=post-commit
----

> * Create the githook secret.

Reviewer: ```suggestion
. Create the Git hook secret.
```

+
----
oc create secret generic githook-ssh-key-secret --from-file=id_rsa=/mypath/.ssh/id_rsa
----

> Expected result: {CENTRAL} pod does not fails even after the githook secret is missing some configurations.

Reviewer: ```suggestion
Expected result: {CENTRAL} pod does not fail even after the Git hook secret is missing some configurations.
```

> Actual result: {CENTRAL} pod fails to start remains stuck at *Container Creating* status.

Reviewer: ```suggestion
Actual result: {CENTRAL} pod fails to start and remains stuck at *Container Creating* status.
```

> Workaround: Specify the following `known_host` file and create the secret again.

Reviewer: ```suggestion
Workaround: Specify the following `known_host` file and create the Git hook secret again.
```
